instrumentation/google.golang.org/grpc/otelgrpc/filters.go

```diff
@@ -0,0 +1,204 @@
+		return !f(i)
+	}
+}
+
+// MethodName returns a Filter that returns true if the request's
+// method name matches the provided string n.
+func MethodName(n string) Filter {
+	return func(i *interceptorInfo) bool {
+		var p gRPCPath
+		switch i.typ {
+		case unaryServer:
+			p = splitFullMethod(i.usinfo.FullMethod)
+		case streamServer:
+			p = splitFullMethod(i.ssinfo.FullMethod)
+		case unaryClient, streamClient:
+			p = splitFullMethod(i.method)
+		default:
+			p = gRPCPath{
+				method: i.method,
+			}
+		}
+		return p.method == n
+	}
+}
+
+// MethodPrefix returns a Filter that returns true if the request's
+// method starts with the provided string pre.
+func MethodPrefix(pre string) Filter {
+	return func(i *interceptorInfo) bool {
+		var p gRPCPath
+		switch i.typ {
+		case unaryServer:
+			p = splitFullMethod(i.usinfo.FullMethod)
+		case streamServer:
+			p = splitFullMethod(i.ssinfo.FullMethod)
+		case unaryClient, streamClient:
+			p = splitFullMethod(i.method)
+		default:
+			p = gRPCPath{
+				method: i.method,
+			}
+		}
+		return strings.HasPrefix(p.method, pre)
+	}
+}
+
+// FullMethodName returns a Filter that returns true if the request's
+// full RPC method string, i.e. /package.service/method, starts with
+// the provided string n.
+func FullMethodName(n string) Filter {
+	return func(i *interceptorInfo) bool {
+		var fm string
+		switch i.typ {
+		case unaryClient, streamClient:
+			fm = i.method
+		case unaryServer:
+			fm = i.usinfo.FullMethod
+		case streamServer:
+			fm = i.ssinfo.FullMethod
+		default:
+			fm = i.method
+		}
+		return fm == n
+	}
+}
+
+// ServiceName returns a Filter that returns true if the request's
+// service name, i.e. package.service, matches s.
+func ServiceName(s string) Filter {
+	return func(i *interceptorInfo) bool {
+		var p gRPCPath
+		switch i.typ {
+		case unaryServer:
+			p = splitFullMethod(i.usinfo.FullMethod)
+		case streamServer:
+			p = splitFullMethod(i.ssinfo.FullMethod)
+		case unaryClient, streamClient:
+			p = splitFullMethod(i.method)
+		default:
+			p = gRPCPath{
+				method: i.method,
+			}
+		}
+		return p.service == s
+	}
+}
+
+// ServicePrefix returns a Filter that returns true if the request's
+// service name, i.e. package.service, starts with the provided string pre.
+func ServicePrefix(pre string) Filter {
+	return func(i *interceptorInfo) bool {
+		var p gRPCPath
+		switch i.typ {
+		case unaryServer:
+			p = splitFullMethod(i.usinfo.FullMethod)
+		case streamServer:
+			p = splitFullMethod(i.ssinfo.FullMethod)
+		case unaryClient, streamClient:
+			p = splitFullMethod(i.method)
+		default:
+			p = gRPCPath{
+				method: i.method,
+			}
+		}
+		return strings.HasPrefix(p.service, pre)
+	}
+}
+
+// NotHealthCheck returns a Filter that returns true if the request's
+// is not health check defined by gRPC Health Checking Protocol.
+// https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+// This is the short of Not(MethodName("grpc.health.v1.Health/Check")).
```

Comment:
```suggestion
```

Seems irrelevant and already inaccurate.  No need to carry documentation details that are evident from the code.